packit/upstream.py

```diff
@@ -162,16 +164,106 @@ def create_patches(
+        logger.debug("sanitized version = %s", ver)
+        return ver
+
+    def bump_spec(self, version: str = None, changelog_entry: str = None):
+        """
+        Run rpmdev-bumpspec on the upstream spec file: it enables
+        schanging version and adding a changelog entry
+
+        :param version: new version which should be present in the spec
+        :param changelog_entry: new changelog entry (just the comment)
+        """
+        cmd = ["rpmdev-bumpspec"]
+        if version:
+            # 1.2.3-4 means, version = 1.2.3, release = 4
+            cmd += ["--new", version]
+        if changelog_entry:
+            cmd += ["--comment", changelog_entry]
+        cmd.append(self.specfile_path)
+        run_command(cmd)
+
+    def create_archive(self):
+        """
+        Create archive, using `git archive` by default,
+        from the content of the upstream repository,
+        only committed changes are present in the archive
+        """
+        if self.package_config.upstream_project_name:
+            dir_name = f"{self.package_config.upstream_project_name}-{self.get_current_version()}"
+        else:
+            dir_name = f"{self.package_name}-{self.get_current_version()}"
+        logger.debug("name + version = %s", dir_name)
+        # We don't care about the name of the archive, really
+        # we just require for the archive to be placed in the cwd
+        if self.package_config.create_tarball_command:
+            archive_cmd = self.package_config.create_tarball_command
+        else:
+            # FIXME: .tar.gz is naive
+            archive_name = f"{dir_name}.tar.gz"
+            archive_cmd = \
+                ["git", "archive", "-o", archive_name, "--prefix", f"{dir_name}/", "HEAD"]
+        run_command(archive_cmd)
+
+    def create_srpm(self, srpm_path: str = None) -> Path:
+        """
+        Create SRPM from the actual content of the repo
+
+        :param srpm_path: path to the srpm
+        :return: path to the srpm
+        """
+        cwd = os.getcwd()
+        cmd = [
+            "rpmbuild", "-bs",
+            "--define", f"_sourcedir {cwd}",
+            "--define", f"_specdir {cwd}",
+            "--define", f"_srcrpmdir {cwd}",
+            # no idea about this one, but tests were failing in tox w/o it
+            "--define", f"_topdir {cwd}",
+            self.specfile_path
+        ]
+        present_srpms = set(Path.cwd().glob("*.src.rpm"))
+        logger.debug("present srpms = %s", present_srpms)
+        out = run_command(
+            cmd, output=True,
+            error_message="SRPM could not be created. Is the archive present?").strip()
+        reg = r"Wrote: (.+)$"
+        try:
+            the_srpm = re.findall(reg, out)[0]
+        except IndexError:
+            raise PackitException("SRPM cannot be found, something is wrong.")
+        if srpm_path:
+            Path(the_srpm).rename(srpm_path)
+            return Path(srpm_path)
+        return Path(the_srpm)
```

Comment:
Dejavu..;-) (I've been implementing that in the old sourcegit/packit...;-)

Reply:
Yeah :D I couldn't find that code, so I wrote it from scratch - my bad I guess.

In SourceGit class, we'll need to change this implementation so it works with generating patches I assume. Or we can have everything in this class.